tgui/packages/tgui/interfaces/Manufacturer/index.tsx

```diff
@@ -0,0 +1,321 @@
+                    >
+                      {data.repeat ? "On" : "Off"}
+                    </LabeledList.Item>
+                    <LabeledList.Item
+                      label="Speed"
+                    >
+                      <Slider
+                        minValue={1}
+                        value={data.speed}
+                        maxValue={data.hacked ? data.max_speed_hacked : data.max_speed_normal}
+                        step={1}
+                        stepPixelSize={100}
+                        disabled={data.mode !== "working"}
+                        onChange={(_e: any, value: number) => act("speed", { "value": value })}
+                      />
+                    </LabeledList.Item>
+                  </LabeledList>
+                </Section>
+                {data.manudrive.limit !== null ? (
+                  <Stack.Item>
+                    <Section
+                      title="Loaded Manudrive"
+                      buttons={
+                        <Button
+                          icon="eject"
+                          content="Eject"
+                          disabled={data.mode !== "ready"}
+                          onClick={() => act("manudrive", { "action": "eject" })}
+                        />
+                      }
+                      mb={1}
+                    >
+                      {data.manudrive.name}
+                      <Divider />
+                      <LabeledList>
+                        <LabeledList.Item
+                          label={"Fabrication Limit"}
+                        >
+                          {data.manudrive.limit === -1 ? "Unlimited" : `${data.manudrive.limit} ${pluralize("use", data.manudrive.limit)}`}
+                        </LabeledList.Item>
+                        {data.manudrive.limit !== -1 ? (
+                          <LabeledList.Item
+                            label={"Remaining Uses"}
+                          >
+                            {data.manudrive_uses_left}
+                          </LabeledList.Item>
+                        ) : null }
+                      </LabeledList>
+                    </Section>
+                  </Stack.Item>
+                ) : null}
+                {data.panel_open ? (
+                  <Stack.Item mb={1} >
+                    <CollapsibleWireMenu wirePanel={wirePanel} />
+                  </Stack.Item>
+                ) : null}
+                <CardInfo />
+              </Stack.Item>
+              <Stack.Item>
+                <Section
+                  title="Rockbox™ Containers"
+                  textAlign="center"
+                >
+                  {data.rockboxes.map((rockbox: Rockbox) => (
+                    <Section
+                      title={rockbox.area_name}
+                      key={rockbox.byondRef}
+                    >
+                      <LabeledList>
+                        {rockbox.ores.length !== 0 ? (rockbox.ores.map((ore: Ore) => (
```

Comment:
```suggestion
                        {rockbox.ores.length !== 0 ? (rockbox.ores.map((ore: OreData) => (
```
I *think* this is coming straight from the `data` variable, so `Ore` here is a `ui_data` (or `ui_static_data`) thing so try to reflect that with use of the `Data` suffix for types. Helps differentiate it from enums, prop types, etc.